File: cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void)
{
    float dollar;
    do
    {
        dollar = get_float("Change owed: ");
    }
    while (dollar < 0);

    int cents = round(dollar * 100);
    int coins = 0;
    int values[4] = {25, 10, 5, 1};

    for (int i = 0; i < 4; i++)
    {
        coins += cents / values[i];
        cents %= values[i];
    }

    printf("%d\n", coins);
}

File: WEEK 6/cash.py
# cash: minimum number of coins

while True:
    try:
        amount = float(input("Change owed: "))
        if amount >= 0:
            break
    except:
        pass

# dollars to cents (avoid floating problems)
cents = round(amount * 100)

coins = 0

# greedy approach
for coin in [25, 10, 5, 1]:
    while cents >= coin:
        cents -= coin
        coins += 1

print(coins)
